app/service/utils.go

```diff
@@ -163,6 +168,41 @@ func toSnakeCase(in string) string {
 	return string(out)
 }
 
+// GetResponseForRouteWithMockedDBAndUser executes a route for unit tests
+// auth.UserIDFromContext is stubbed to return the given userID.
+// The test should provide functions that prepare the router and the sql mock
+func GetResponseForRouteWithMockedDBAndUser(
+	method string, path string, requestBody string, userID int64,
+	setMockExpectationsFunc func(sqlmock.Sqlmock),
+	setRouterFunc func(router *chi.Mux, baseService *Base)) (*http.Response, sqlmock.Sqlmock, string, error) {
+
+	logs := setupLogsCaptureForTests()
+	defer monkey.UnpatchAll()
+
+	db, mock := database.NewDBMock()
+	defer func() { _ = db.Close() }() // nolint: gosec
+
+	setMockExpectationsFunc(mock)
+
+	base := Base{Store: database.NewDataStore(db), Config: nil}
+	router := chi.NewRouter()
+	setRouterFunc(router, &base)
+
+	monkey.Patch(auth.UserIDFromContext, func(context context.Context) int64 {
+		return userID
```

Comment:
And you do not remove the patch after it then 🤨 . So you do it when it is trivial, not in other case ? ;-) (okay okay, probably I do the same 😉)
Again here as it is not an external uncontrollable package (such as time or random), we should do it properly, it should be quite easy actually. Let me try.